universal-application-tool-0.0.1/app/services/question/QuestionServiceImpl.java

```diff
@@ -137,6 +137,15 @@ public void discardDraft(Long id) throws InvalidUpdateException {
       throw new InvalidUpdateException("Did not find question in draft version.");
     }
     question.get().save();
+
+    // Update any repeated questions that may have referenced the discarded question.
+    Version activeVersion = versionRepositoryProvider.get().getActiveVersion();
+    Optional<Question> activeQuestion =
+        activeVersion.getQuestionByName(question.get().getQuestionDefinition().getName());
+    questionRepository.updateAllRepeatedQuestions(
```

Comment:
should we check if they are equal first before updating them?

Reply:
I waffled but went ahead an added it out of caution as it implies data corruption; so that at least we're clearly aware of it.

The same revision ID for a question can't be ACTIVE and DRAFT, and it was checked as being a valid draft ID so this state would indicate data corruption if it happened.

Reply:
ooo okay gotcha, i mean if we ant to handle it in the deletion section and not here that's also fine by me was just curious

Reply:
Not sure what you mean, but it's best to assert everything is as expected before making partial changes.